codejail/jail.py

```diff
@@ -0,0 +1,298 @@
+"""
+Core Jail class and supporting code
+"""
+
+from collections import namedtuple
+import json
+import logging
+import os
+import shutil
+
+from .exceptions import JailError, SafeExecException
+from . import languages
+from . import limits
+from .proxy import run_subprocess_through_proxy
+from .subproc import run_subprocess
+from .util import json_safe, temp_directory
+
+log = logging.getLogger("codejail")
+
+
+JailResult = namedtuple('JailResult', ['status', 'stdout', 'stderr'])
+
+
+class Jail(object):
+    """
+    Self contained codejail context.
+    """
+    default_language = languages.other
+
+    known_languages = {
+        'python': languages.python2,
+    }
```

Comment:
Maybe this should be `known_commands` instead of `known_languages`, since it's really saying that we have special default behavior for the command named `"python"`, not that the language `languages.python2` is special.